reconcile/ocm_clusters.py

```diff
@@ -24,6 +24,42 @@ def fetch_desired_state(clusters):
     return desired_state
 
 
+def get_cluster_update_spec(cluster_name, current_spec, desired_spec):
+    """ Get a cluster spec to update. Returns an error if diff is invalid """
+    allowed_spec_update_fields = [
```

Comment:
Make this a set

Reply:
why would a set be better?

Reply:
What you do below logicallly the difference of the keys of your dictionary and the allowed keys (my first revision stated intersection because I was on a hurry and sleep deprived). You can express the operation with one line of code instead of a loop and an `if`. Then, the cost of the entire difference is O(N) (or O(N*log N), depending on internal implementation details). The cost of each an `in` for a list is O(N), which makes the loop below O(N^2).

I don't care much about speed here, but when you add the reduced code size and the readability, sets are totally the way to go here.